app/assets/javascripts/govuk_publishing_components/components/single-page-notification-button.js

```diff
@@ -28,7 +29,12 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
           // test that the html returned contains the button component; if yes, swap the button for the updated version
           var responseHasButton = html.querySelector('form.gem-c-single-page-notification-button .gem-c-single-page-notification-button__submit')
           if (responseHasButton) {
-            this.$module.outerHTML = newButton
+            var parent = this.$module.parentElement
+            var updatedModule = parent.insertBefore(html.firstChild, this.$module)
+            parent.removeChild(this.$module)
```

Comment:
I'm slightly nervous about removing `this.$module` entirely - although it looks like the module doesn't do anything else after this. Is that correct?

Reply:
Yes the module does not do anything after this. The whole point of this script is solely to get the button to communicate accurately whether the current user has an active subscription to the current page.